### DISCUSSION TITLE
openslide.mpp-x、openslide.objective-power etc.

### DISCUSSION THREAD

**Discussion by yanboma** [libvips/libvips#3410] [General]
how to get "openslide.mpp-x、openslide.objective-power" and other information through interface?

The information gotted by the xres() interface is quite different from that of mpp-x

**Comment by jcupitt** [comment]
Hey @yanboma,

It should be the same I think, it's just that the units are different. xres/yres are pixels per millimetre, and the openslide mpp fields are microns per pixel, so they are reciprocals of each other.

For example:

```
$ vipsheader -f openslide.mpp-x ~/pics/openslide/CMU-1.svs
0.499
$ vipsheader -f xres ~/pics/openslide/CMU-1.svs
2004.0080160320642
```

0.499 microns per pixel means 1/0.499 pixels per micron, or 2.004008. There are 1000 microns in a millimetre, so that's 2004.008 pixels per mm.

**Reply by jcupitt** [reply]
... or in python:

```
$ python3
Python 3.10.7 (main, Mar 10 2023, 10:47:39) [GCC 12.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import pyvips
>>> x = pyvips.Image.new_from_file("CMU-1.svs")
>>> x.get("openslide.mpp-x")
'0.499'
>>> 
```

Note that openslide properties are strings, so you'll need to parse that to get a float.
